### ISSUE TITLE
`Error: database is locked` when running migrations  on an existing Sqlite DB

### ISSUE THREAD

**Issue by mindreframer** [swiftcarrot/queryx#38]
```bash
$ ./queryx db:migrate --schema schema.hcl
run 20230816163454 db/migrations/20230816163454_auto.up.sql
Error: database is locked
Usage:
  queryx db:migrate [flags]

Flags:
  -h, --help   help for db:migrate

Global Flags:
      --schema string   queryx schema file (default "schema.hcl")

database is locked
```

Removing the DB file and running migrations on a clean slate works without issues.

**Comment by wangzuo** [comment]
The sqlite database is being used by another process rather than queryx?

**Comment by wangzuo** [comment]
@mindreframer feel free to reopen if it still remains.

**Comment by mindreframer** [comment]
@wangzuo The issue still remains. The database is not open otherwise. Just reproduced with queryx_v0.2.3. 

For reproduction: 
cat schema.hlc
```hlc
database "db" {
  adapter   = "sqlite"

  config "development" {
    url = "sqlite:dev.sqlite3"
  }

  model "Client" {
    default_primary_key = false
    timestamps          = false

    column "name" {
      type = string
    }
    column "float" {
      type = float
    }
    column "added" {
      type = string
    }
  }
}
```

Run 

```bash 
$ ./queryx db:migrate --schema schema.hcl
Created db/migrations/20230824181524_auto.up.sql
Created db/migrations/20230824181524_auto.down.sql
run 20230824181524 db/migrations/20230824181524_auto.up.sql
```

Now add a column, like: 
```
   column "added2" {
      type = string
    }
```

Run again: 
```
$ ./queryx db:migrate --schema schema.hcl

Created db/migrations/20230824181841_auto.up.sql
Created db/migrations/20230824181841_auto.down.sql
run 20230824181841 db/migrations/20230824181841_auto.up.sql
Error: database is locked
Usage:
  queryx db:migrate [flags]

Flags:
  -h, --help   help for db:migrate

Global Flags:
      --schema string   queryx schema file (default "schema.hcl")

database is locked
```


Migraiton was not executed, and the DB was not open otherwise.

**Comment by wangzuo** [comment]
@mindreframer This is due to an unclosed sql.rows. We have added a test in github action and released a fix in v0.2.4. Many thanks and feel free to reopen it if still happens.
